geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSInterStoreIntegrationTest.java

```diff
@@ -0,0 +1,237 @@
+    jedis.sadd(SET_KEY_2, SET_MEMBERS_1);
+
+    assertThat(jedis.sinterstore(DESTINATION_KEY, SET_KEY_1, SET_KEY_2))
+        .isEqualTo(SET_MEMBERS_1.length);
+    assertThat(jedis.smembers(DESTINATION_KEY)).containsExactlyInAnyOrder(SET_MEMBERS_1);
+  }
+
+  @Test
+  public void sinterstore_withMultipleNonExistentSets_returnsZero_destKeyDoesNotExist() {
+    assertThat(jedis.sinterstore(DESTINATION_KEY, NON_EXISTENT_SET, "{tag1}nonExistentSet2"))
+        .isEqualTo(0);
+    assertThat(jedis.exists(DESTINATION_KEY)).isFalse();
+  }
+
+  @Test
+  public void sinterstore_withExistentSet_returnsInterSize_destKeyOverwrittenWithInter() {
+    jedis.sadd(DESTINATION_KEY, DESTINATION_MEMBERS);
+    jedis.sadd(SET_KEY_1, SET_MEMBERS_1);
+    assertThat(jedis.sinterstore(DESTINATION_KEY, SET_KEY_1)).isEqualTo(SET_MEMBERS_1.length);
+    assertThat(jedis.smembers(DESTINATION_KEY)).containsExactlyInAnyOrder(SET_MEMBERS_1);
+  }
+
+  @Test
+  public void sinterstore_withNonExistentSet_returnsZero_destKeyIsDeleted() {
+    jedis.sadd(DESTINATION_KEY, DESTINATION_MEMBERS);
+    assertThat(jedis.sinterstore(DESTINATION_KEY, NON_EXISTENT_SET)).isEqualTo(0);
+    assertThat(jedis.exists(DESTINATION_KEY)).isFalse();
+  }
+
+  @Test
+  public void sinterstore_withNonSetDestKey_withExistentSet_returnsInterSize_destKeyOverwrittenWithInter() {
+    String stringKey = "{tag1}ding";
+    jedis.set(stringKey, "dong");
+
+    jedis.sadd(SET_KEY_1, SET_MEMBERS_1);
+    assertThat(jedis.sinterstore(stringKey, SET_KEY_1)).isEqualTo(SET_MEMBERS_1.length);
+    assertThat(jedis.smembers(stringKey)).containsExactlyInAnyOrder(SET_MEMBERS_1);
+  }
+
+  @Test
+  public void sinterstore_withNonSetDestKey_withNonExistentSet_returnsZero_destKeyIsDeleted() {
+    String stringKey = "{tag1}ding";
+    jedis.set(stringKey, "dong");
+
+    assertThat(jedis.sinterstore(stringKey, NON_EXISTENT_SET)).isEqualTo(0);
+    assertThat(jedis.exists(DESTINATION_KEY)).isFalse();
+  }
+
+  @Test
+  public void sinterstore_withNonExistentDest_withNonSetKeyAsFirstKey_returnsWrongTypeError() {
+    String stringKey = "{tag1}ding";
+    jedis.set(stringKey, "dong");
+
+    jedis.sadd(SET_KEY_1, SET_MEMBERS_1);
+    jedis.sadd(SET_KEY_2, "doorbell");
+    assertThatThrownBy(() -> jedis.sinterstore(DESTINATION_KEY, stringKey, SET_KEY_1, SET_KEY_2))
+        .hasMessageContaining(ERROR_WRONG_TYPE);
+  }
+
+  @Test
+  public void sinterstore_withExistentDest_withNonSetKeyAsThirdKey_returnsWrongTypeError() {
+    String stringKey = "{tag1}ding";
+    jedis.set(stringKey, "dong");
+
+    jedis.sadd(DESTINATION_KEY, DESTINATION_MEMBERS);
+    jedis.sadd(SET_KEY_1, SET_MEMBERS_1);
+    jedis.sadd(SET_KEY_2, "doorbell");
+    assertThatThrownBy(() -> jedis.sinterstore(DESTINATION_KEY, SET_KEY_1, SET_KEY_2, stringKey))
+        .hasMessageContaining(ERROR_WRONG_TYPE);
+  }
+
+  @Test
+  public void sinterstore_withNonExistentDest_withNonSetKeyAsThirdKeyAndNonExistentSetAsFirstKey_returnsWrongTypeError() {
```

Comment:
Just to simplify it a little, the "withNonExistentDest" part of this test name can be removed, I think, since we don't expect that whether the destination exists or not will effect the behavviour here.